Automattic/jetpack
# Abstract button block width settings in a new `useWidth` hook

Required by https://github.com/Automattic/jetpack/pull/25297

#### Changes proposed in this Pull Request:

This PR abstracts the button width settings in a new `useWidth` hook so they can be used by any other block with a `width` attribute. This will helps us to implement a Payment buttons block container since we'll need to control the width of the inner Payment button blocks.

This was initially implemented in https://github.com/Automattic/jetpack/pull/25297, but I decided to open a separate PR to facilitate the review.

#### Other information:


#### Jetpack product discussion
No.

#### Does this pull request change what data or activity we track or use?
N/A.

#### Testing instructions:

Since this PR doesn't introduce any behavior or aesthetic change, you mainly need to double check that there are no regressions for any block that uses the Button block.

- [Spin up a Jurassic Ninja site running this branch](https://jurassic.ninja/create/?jetpack-beta&branches.jetpack=update/payment-button-block-width).
- Set up Jetpack.
- Go to `/wp-admin/post-new.php`.
- Insert a Payment Button block.
- Make sure you can change the width of the button as usual.
- Insert a Form block.
- Choose Contact Form.
- Make sure you can change the width of the submit button as usual.
